### ISSUE TITLE
ensure_xcode_version crashes with fastlane 2.0.2

### ISSUE THREAD

**Issue by justin** [fastlane/fastlane#7518]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

I updated us to v2.0 today and we are getting a crash in our script when we check `ensure_xcode_version(version: "8.2")` in one of our lanes. Error is below.

### Captured Output

Command Used: `beta`
<details><summary>Output/Log</summary>

```

[16:51:46]: -------------------------------------------------
[16:51:46]: --- Step: Verifying required fastlane version ---
[16:51:46]: -------------------------------------------------
[16:51:46]: Your fastlane version 2.0.2 matches the minimum requirement of 2.0.2  ✅
[16:51:46]: ------------------------------
[16:51:46]: --- Step: default_platform ---
[16:51:46]: ------------------------------
[16:51:46]: Driving the lane 'ios beta' 🚀
[16:51:46]: --------------------------------
[16:51:46]: --- Step: get_version_number ---
[16:51:46]: --------------------------------
[16:51:46]: $ cd /Users/justin/Development/TEDapp && agvtool what-marketing-version -terse
[16:51:47]: ▸ "TED.xcodeproj/../TED-Info.plist"=3100
[16:51:47]: ▸ "TED.xcodeproj/../TEDXCTests/Info.plist"=1.0
[16:51:47]: ------------------------------
[16:51:47]: --- Step: get_build_number ---
[16:51:47]: ------------------------------
[16:51:47]: $ cd /Users/justin/Development/TEDapp && agvtool what-version -terse
[16:51:48]: ▸ 2581
[16:51:48]: --------------------------------------
[16:51:48]: --- Step: Switch to ios build lane ---
[16:51:48]: --------------------------------------
[16:51:48]: Cruising over to lane 'ios build' 🚖
[16:51:48]: ----------------------------------
[16:51:48]: --- Step: ensure_xcode_version ---
[16:51:48]: ----------------------------------
[16:51:48]: $ xcversion selected
[16:51:48]: ▸ /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship/client.rb:22:in `<class:Client>': uninitialized constant Spaceship::Client::Fastlane (NameError)
[16:51:48]: ▸ Did you mean?  FALSE
[16:51:48]: ▸ from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship/client.rb:20:in `<module:Spaceship>'
[16:51:48]: ▸ from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship/client.rb:19:in `<top (required)>'
[16:51:48]: ▸ from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship.rb:2:in `require'
[16:51:48]: ▸ from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship.rb:2:in `<top (required)>'
[16:51:48]: ▸ from /Users/justin/.gem/gems/xcode-install-2.0.9/lib/xcode/install.rb:4:in `require'
[16:51:48]: ▸ from /Users/justin/.gem/gems/xcode-install-2.0.9/lib/xcode/install.rb:4:in `<top (required)>'
[16:51:48]: ▸ from /Users/justin/.gem/gems/xcode-install-2.0.9/bin/xcversion:10:in `require'
[16:51:48]: ▸ from /Users/justin/.gem/gems/xcode-install-2.0.9/bin/xcversion:10:in `<top (required)>'
[16:51:48]: ▸ from /Users/justin/.gem/bin/xcversion:23:in `load'
[16:51:48]: ▸ from /Users/justin/.gem/bin/xcversion:23:in `<main>'
+------------------+----------+
|        Lane Context         |
+------------------+----------+
| DEFAULT_PLATFORM | ios      |
| PLATFORM_NAME    | ios      |
| LANE_NAME        | ios beta |
| VERSION_NUMBER   | 3100     |
| BUILD_NUMBER     | 2581     |
+------------------+----------+
[16:51:48]: Exit status of command 'xcversion selected' was 1 instead of 0.
/Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship/client.rb:22:in `<class:Client>': uninitialized constant Spaceship::Client::Fastlane (NameError)
Did you mean?  FALSE
	from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship/client.rb:20:in `<module:Spaceship>'
	from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship/client.rb:19:in `<top (required)>'
	from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship.rb:2:in `require'
	from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship.rb:2:in `<top (required)>'
	from /Users/justin/.gem/gems/xcode-install-2.0.9/lib/xcode/install.rb:4:in `require'
	from /Users/justin/.gem/gems/xcode-install-2.0.9/lib/xcode/install.rb:4:in `<top (required)>'
	from /Users/justin/.gem/gems/xcode-install-2.0.9/bin/xcversion:10:in `require'
	from /Users/justin/.gem/gems/xcode-install-2.0.9/bin/xcversion:10:in `<top (required)>'
	from /Users/justin/.gem/bin/xcversion:23:in `load'
	from /Users/justin/.gem/bin/xcversion:23:in `<main>'


+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | get_version_number                  | 0           |
| 4    | get_build_number                    | 0           |
| 5    | Switch to ios build lane            | 0           |
| 💥   | ensure_xcode_version                | 0           |
+------+-------------------------------------+-------------+

[16:51:48]: fastlane finished with errors

[!] Exit status of command 'xcversion selected' was 1 instead of 0.
/Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship/client.rb:22:in `<class:Client>': uninitialized constant Spaceship::Client::Fastlane (NameError)
Did you mean?  FALSE
	from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship/client.rb:20:in `<module:Spaceship>'
	from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship/client.rb:19:in `<top (required)>'
	from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship.rb:2:in `require'
	from /Users/justin/.gem/gems/fastlane-2.0.2/spaceship/lib/spaceship.rb:2:in `<top (required)>'
	from /Users/justin/.gem/gems/xcode-install-2.0.9/lib/xcode/install.rb:4:in `require'
	from /Users/justin/.gem/gems/xcode-install-2.0.9/lib/xcode/install.rb:4:in `<top (required)>'
	from /Users/justin/.gem/gems/xcode-install-2.0.9/bin/xcversion:10:in `require'
	from /Users/justin/.gem/gems/xcode-install-2.0.9/bin/xcversion:10:in `<top (required)>'
	from /Users/justin/.gem/bin/xcversion:23:in `load'
	from /Users/justin/.gem/bin/xcversion:23:in `<main>'



```

</details>

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.12.2                                     |
| Ruby                | 2.3.3                                       |
| Bundler?            | true                                        |
| Git                 | git version 2.11.0                          |
| Installation Source | ~/.gem/bin/fastlane                         |
| Host                | Mac OS X 10.12.2 (16C67)                    |
| Ruby Lib Dir        | /usr/local/Cellar/ruby/2.3.3/lib            |
| OpenSSL Version     | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained        | false                                       |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_version "2.0.2"
default_platform :ios

platform :ios do
  # Store the current CFBundleVersion and CFBundleShortVersionString in the fastlane context before every lane.
  before_all do
    Actions.lane_context[SharedValues::VERSION_NUMBER] = get_version_number
    Actions.lane_context[SharedValues::BUILD_NUMBER] = get_build_number
  end

  desc "Run iOS Test Suite"
  lane :test do |options|
    scan
  end

  desc "Swift Lint"
  lane :lint do
    swiftlint
  end

  desc "Submit a new Beta Build to TestFlight"
  lane :beta do
    build

    changelog = prompt(
      text: "Changelog: ",
      multi_line_end_keyword: "END"
    )

    pilot(
      changelog: changelog,
      team_name: ENV['PILOT_TEAM_NAME'],
      team_id: ENV['PILOT_TEAM_ID'],
      skip_submission: true,
      skip_waiting_for_build_processing:true
    )
    clean_build_artifacts
  end

  desc "Deploy a new version to the App Store"
  lane :release do
    build(release: true)
    deliver(force: true)
    clean_build_artifacts
  end

  desc "Download the Bitcode dSYMs for the most recent TestFlight build"
  lane :refresh_dsyms do
    build_number = lane_context[SharedValues::BUILD_NUMBER]
    download_dsyms(build_number: build_number)

    clean_build_artifacts
  end

  after_all do |lane|
    notification(message: "Fastlane '#{lane}' successfully finished.")
  end

  # ====================================
  # Private Lanes
  # ====================================
  private_lane :build do |options|
    release_type =  options[:release] ? "Store" : "Beta"

    ensure_xcode_version(version: "8.2")
    ensure_git_status_clean
    new_build_number = increment_build_number
    sigh(output_path: "./Certificates & Profiles/")
    ENV["PROFILE_UUID"] = lane_context[SharedValues::SIGH_UDID]
    gym

    commit_version_bump(
      xcodeproj: 'TED.xcodeproj',
      message: "Build Increment: " + new_build_number + "\n\n[ci skip]",
      force: true
    )

    push_to_git_remote

    # Only Tag if we are doing a release build.
    if options[:release]
      version = app_version
      add_git_tag tag: "v" + version
      push_git_tags
      archive
    end
  end

  private_lane :app_version do |options|

    plist_file_path = sh("xcodebuild -project ../TED.xcodeproj -showBuildSettings | grep INFOPLIST_FILE | sed 's/[ ]*INFOPLIST_FILE = //'").chomp

    version = get_info_plist_value(
      key: "TEDAppStoreVersionString",
      path: plist_file_path
    )

    build = get_build_number
    version = version
    puts "Working with App version: " + version
    puts "Working with App build: " + build
    version
  end

  private_lane :archive do |options|
    archive = lane_context[SharedValues::XCODEBUILD_ARCHIVE]
    version = app_version
    output = "/tmp"

    backup_xcarchive(
      xcarchive: archive,
      destination: output,
      zip: false,
      versioned: false
    )

    name = "TED-#{version}.zip"
    xcarchive = lane_context[SharedValues::BACKUP_XCARCHIVE_FILE]
    zipped = zip(
      path: xcarchive,
      output_path: "~/Desktop/#{name}"
    )

    FileUtils.rm_rf(xcarchive)

    upload_to_s3(
      file: zipped,
      remote_path: "Releases/com.ted.TED/#{name}"
    )
  end
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# The Appfile stores useful information that are used across all fastlane tools like your
# Apple ID or the application Bundle Identifier, to deploy your lanes faster and tailored
# on your project needs.
# https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

app_identifier "com.ted.TED"
apple_id ENV['FASTLANE_USERNAME']

team_name "TED Conferences"
team_id "4Z6E862WH7"
```
</details>

### fastlane gems

| Gem                 | Version | Update-Status |
| ------------------- | ------- | ------------- |
| credentials_manager | 0.16.2  | ✅ Up-To-Date  |
| spaceship           | 0.39.0  | ✅ Up-To-Date  |
| fastlane            | 2.0.2   | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.2        |
| bundler                   | 1.13.6       |
| io-console                | 0.4.6        |
| CFPropertyList            | 2.3.4        |
| i18n                      | 0.7.0        |
| json                      | 1.8.3        |
| minitest                  | 5.10.1       |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| activesupport             | 4.2.7.1      |
| public_suffix             | 2.0.4        |
| addressable               | 2.5.0        |
| aws-sigv4                 | 1.0.0        |
| jmespath                  | 1.3.1        |
| aws-sdk-core              | 2.6.38       |
| aws-sdk-resources         | 2.6.38       |
| aws-sdk                   | 2.6.38       |
| babosa                    | 1.0.2        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander                 | 4.4.1        |
| security                  | 0.1.3        |
| credentials_manager       | 0.16.2       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| dotenv                    | 2.1.1        |
| excon                     | 0.54.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.10.0       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.10.1       |
| fastimage                 | 2.0.1        |
| gh_inspector              | 1.0.2        |
| jwt                       | 1.5.6        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| hurley                    | 0.2          |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| google-api-client         | 0.9.20       |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.2.0        |
| rubyzip                   | 1.2.0        |
| slack-notifier            | 1.5.1        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.2        |
| terminal-table            | 1.7.3        |
| word_wrap                 | 1.0.0        |
| xcode-install             | 2.0.9        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| xcpretty-travis-formatter | 0.0.4        |
</details>

*generated on:* **2016-12-15**
</details>

**Comment by KrauseFx** [comment]
Thanks for submitting this issue. We'll have a fic ready as soon as possible!

**Comment by milch** [comment]
I just released [`2.0.3`](https://github.com/fastlane/fastlane/releases/tag/2.0.3) which fixes this issue. Let me know if it helps! 🚀
